include/seqan3/alignment/matrix/detail/alignment_score_matrix_one_column_banded.hpp

```diff
@@ -0,0 +1,194 @@
+    constexpr alignment_score_matrix_one_column_banded(first_sequence_t && first,
+                                                       second_sequence_t && second,
+                                                       static_band const & band,
+                                                       score_t const initial_value = score_t{})
+    {
+        matrix_base_t::cols = static_cast<size_t>(std::ranges::distance(first) + 1);
+        matrix_base_t::rows = static_cast<size_t>(std::ranges::distance(second) + 1);
+
+        band_col_index = std::min<int32_t>(std::max<int32_t>(band.upper_bound, 0), matrix_base_t::cols - 1);
+        band_row_index = std::min<int32_t>(std::abs(std::min<int32_t>(band.lower_bound, 0)), matrix_base_t::rows - 1);
+        band_size = band_col_index + band_row_index + 1;
+
+        // Reserve one more cell to deal with last cell in the banded column which needs only the diagonal and up cell.
+        matrix_base_t::pool.resize(band_size + 1, element_type{initial_value, initial_value});
+    }
+    //!\}
+
+private:
+
+    //!\brief The column index where the upper bound of the band passes through.
+    int32_t band_col_index{};
+    //!\brief The row index where the lower bound of the band passes through.
+    int32_t band_row_index{};
+    //!\brief The size of the band.
+    int32_t band_size{};
+
+    //!\copydoc seqan3::detail::alignment_matrix_column_major_range::initialise_column
+    constexpr alignment_column_type initialise_column(size_t const column_index) noexcept
+    {
+        int32_t slice_begin = std::max<int32_t>(0, band_col_index - column_index);
+        int32_t row_end_index = column_index - band_col_index + band_size;
+        int32_t slice_end = band_size - std::max<int32_t>(row_end_index - matrix_base_t::rows, 0);
+
+        assert(row_end_index >= 0);
+        assert(slice_begin >= 0);
+        assert(slice_end > 0);
+        assert(slice_begin < slice_end);
+
+        return alignment_column_type{*this, column_data_view_type{std::addressof(matrix_base_t::pool[slice_begin]),
+                                                                  std::addressof(matrix_base_t::pool[slice_end])}};
+    }
+
+    //!\copydoc seqan3::detail::alignment_matrix_column_major_range::make_proxy
+    template <std::RandomAccessIterator iter_t>
+    constexpr value_type make_proxy(iter_t host_iter) noexcept
+    {
+        return {std::get<0>(*host_iter),             // current
+                std::get<0>(matrix_base_t::cache),   // last diagonal
+                std::get<1>(*(host_iter + 1)),       // last left (read)
+                std::get<1>(*(host_iter)),           // next left (write)
+                std::get<1>(matrix_base_t::cache)};  // last up
+    }
+
+    //!\copydoc seqan3::detail::alignment_matrix_column_major_range::on_column_iterator_creation
+    template <std::RandomAccessIterator iter_t>
+    constexpr void on_column_iterator_creation(iter_t host_iter) noexcept
```

Comment:
That is not the iterator over the column but basically an iterator type over the view that is generated in `initialise_column`. Therefor it would be ok to get to the iterator type in the derived class but this way it is much more practical. Again there is no generic use case for this but a clear dependency between the derived and the base class